superset-frontend/src/SqlLab/components/QueryTable/index.jsx

```diff
@@ -98,6 +115,36 @@ const QueryTable = props => {
           q.duration = fDuration(q.startDttm, q.endDttm);
         }
         const time = moment(q.startDttm).format().split('T');
+        const statusConfig = {
+          name: '',
+          label: '',
+          status: '',
+        };
+        if (q.state === 'success') {
```

Comment:
To simplify some of the code here and where you set the color you could move all of this to a map:

```js
statusAttributes = {
  success: {
    color: theme.colors.success.base,
    config: {
      name: 'check',
      label: t('Success'),
      status: 'success',
    },
  },
  ...
}
```

Then you can get the color with:

```js
color: ${({ status, theme }) => statusAttributes[status]?.color || theme.colors.grayscale.base;
```

And here:

```js
statusConfig = statusAttributes[q.state]?.config || defaultStatusConfig;
```